examples/fluids/src/turb_spanstats.c

```diff
@@ -43,19 +43,20 @@ PetscErrorCode CreateStatsDM(User user, ProblemData *problem, PetscInt degree) {
   user->spanstats.span_width = domain_max[2] - domain_min[1];
 
   {  // Get DM from surface
-    DM          parent_distributed_dm;
-    PetscSF     isoperiodicface;
-    DMLabel     label;
-    PetscMPIInt size;
+    DM             parent_distributed_dm;
+    const PetscSF *isoperiodicface;
+    PetscInt       num_isoperiodicface;
+    DMLabel        label;
+    PetscMPIInt    size;
 
-    PetscCall(DMPlexGetIsoperiodicFaceSF(user->dm, &isoperiodicface));
+    PetscCall(DMPlexGetIsoperiodicFaceSF(user->dm, &num_isoperiodicface, &isoperiodicface));
 
     if (isoperiodicface) {
       PetscSF         inv_isoperiodicface;
       PetscInt        nleaves;
       const PetscInt *ilocal;
 
-      PetscCall(PetscSFCreateInverseSF(isoperiodicface, &inv_isoperiodicface));
+      PetscCall(PetscSFCreateInverseSF(isoperiodicface[num_isoperiodicface - 1], &inv_isoperiodicface));
```

Comment:
This sounds like an environment assumption that the last isoperiodic dimension is the one on which to do spanwise averaging. Can we make it explicit?